portfolio/src/main/webapp/map.js

```diff
@@ -12,44 +12,50 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-const API_KEY = "AIzaSyAxYeZWtbYyULIb5s59BaENaSpZD2Qr3RA";
+const API_KEY = "AIzaSyDOLA6fsSWwR-ONnnzTR0FnjxWhXc33wJw";
 
 function loadPage() {
-  createMap();
-  renderAddressForm();
+  renderMarkers(createMap());
+  activateAddressForm();
 }
 
-function renderAddressForm() {
+function activateAddressForm() {
   // On submit, address-form will convert city, state to lat, lng and POST that to datastore
   document.getElementById('address-form').addEventListener('submit', (e) => {
+    // Need to preventDefault() so page doesn't refresh immediately.
+    e.preventDefault();
     const formData = new FormData(e.target);
     const city = formData.get("city").replace(' ', '+');
     const state = formData.get("state").replace(' ', '+');
     fetch("https://maps.googleapis.com/maps/api/geocode/json?address=" + city + "," + state + "&key=" + API_KEY)
```

Comment:
Ideally this request to the geocoding API should be made from your server after receiving a request from you client. Normally most things that require an API key would be on the server and you would avoid submitting the API key to GitHub because it can be stolen and used by other people (but you end up paying for the API use). I'm not sure if this is necessarily required for the geocoding API though because you can restrict where it's used from. Please look into this and at least restrict the use of this key appropriately.

If you followed a guide to get his code (and it said to put it in the JavaScript) it's probably fine.